DataManagementSystem/Service/StorageElementHandler.py

```diff
@@ -74,420 +75,503 @@ def getDiskSpace(path, size = 'TB', total = False):
       # return free space
       queried_size = st.f_bavail
 
-    result = float( queried_size * st.f_frsize ) / float(convert)
+    result = float(queried_size * st.f_frsize) / float(convert)
 
   except OSError as e:
-    return S_ERROR( errno.EIO, "Error while getting the available disk space: %s" % repr(e) )
+    return S_ERROR(errno.EIO, "Error while getting the available disk space: %s" % repr(e))
+
+  return S_OK(round(result, 4))
+
+
+def getTotalDiskSpace():
+  """  Returns the total maximum volume of the SE storage in MB. The total volume
+       can be limited either by the amount of the available disk space or by the
+       MAX_STORAGE_SIZE value.
+
+       :return: S_OK/S_ERROR, Value is the max total volume of the SE storage in MB
+  """
+
+  global BASE_PATH
+  global MAX_STORAGE_SIZE
+
+  result = getDiskSpace(BASE_PATH, size='MB', total=True)
+  if not result['OK']:
+    return result
+  totalSpace = result['Value']
+  maxTotalSpace = min(totalSpace, MAX_STORAGE_SIZE) if MAX_STORAGE_SIZE else totalSpace
+  return S_OK(maxTotalSpace)
+
 
-  return S_OK( round(result, 4) )
+def getFreeDiskSpace(ignoreMaxStorageSize=True):
+  """ Returns the free disk space still available for writing taking into account
+      the total available disk space and the MAX_STORAGE_SIZE limitation
 
-def initializeStorageElementHandler( serviceInfo ):
+      :return: S_OK/S_ERROR, Value is the free space of the SE storage in MB
+  """
+
+  global MAX_STORAGE_SIZE
+  global BASE_PATH
```

Comment:
Do we really need all these globals? They are considered a not-so-nice practice.

Reply:
I just continue the same style as was in the original file. Re-styling it is not the goal of this PR